syntax = "proto3";

service CacheService {
  rpc GetValue(KeyRequest) returns (ValueResponse);
  rpc SetValue(KeyValue) returns (SetResponse);
}

message KeyRequest {
  string key = 1;
}

message ValueResponse {
  string value = 1;
  bool found = 2;
}

message KeyValue {
  string key = 1;
  string value = 2;
}

message SetResponse {
  bool success = 1;
}
